src/main/java/org/dependencytrack/policy/vulnerability/VulnerabilityPolicy.java

```diff
@@ -0,0 +1,108 @@
+package org.dependencytrack.policy.vulnerability;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+
+public class VulnerabilityPolicy {
```

Comment:
I intentionally duplicated this class from the persistence model introduced in #462. The intention is that the policy evaluator should not have to care where the policies are coming from. In case we want this to be an SPI in the future, it should also not depend on any internal model definitions.

Curious about opinions on these two approaches:

* Implementations of `VulnerabilityPolicyProvider` map their internal model to this `VulnerabilityPolicy` class
* We make `VulnerabilityPolicy` an interface instead, and the internal model of the `VulnerabilityPolicyProvider` simply implements that

The former makes it easier to deal with breaking changes, the latter means that providers do not need to do any mapping. Thoughts?

Reply:
If I was a provider, I usually get caught up in mapping because of doubts on what certain fields mean. Interface appears cleaner approach. But when we are provider also, first approach could be quickest to implement.

Reply:
I was just about to change it to an interface, but then noticed that by doing it this way, we also have a hard coupling of data types. Say for `validFrom`, the database model uses `java.util.Date`:

https://github.com/DependencyTrack/hyades-apiserver/blob/fd958b2967f0fe1f0b8ae6f5a22da3ff19b8b393/src/main/java/org/dependencytrack/model/VulnerabilityPolicy.java#L64-L66

but the `VulnerabilityPolicy` interface may use the newer `java.time.Instant` instead.

If the interface defines a method `getValidFrom`, then our internal data model would have to implement this method, using the data type dictated by the interface. The ORM and JSON mappers use getters and setters to access fields, so a constellation like:

```java
private Date validFrom;

@Override // Implements VulnerabilityAnalysis interface
public Instant getValidFrom() {}

public void setValidFrom(Date validFrom) {}
```

could be problematic. Just using `Date` and `Instant` as examples, might just as well be other data types. Limitations of the ORM should not dictate what data types we can use in a (potentially) public SPI.